### PR TITLE
Custom field translations

### PR DESCRIPTION
**Pull request by cconard96**
## Checklist before requesting a review

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.

## Description

Allow translating custom field labels

### CODE REVIEW

**Comment by cconard96** [comment]
Waiting for functional/technical review to ensure this feature is finalized before adding tests.

**Comment by orthagh** [comment]
Ok, I was curious about where the translation system will be displayed.
Re-using the existing tab (translations) is clever in fact.

I suggest the following changes:
- remove the "Custom field translation"
- Always display the "Field" dropdown
- include the "asset name" option in this dropdown
- add a helper on the top of the modal explaining you can translate either the asset name or custom fields.
- merge the 2 tables into the second one, translation entry for the asset name displayed in the single list, and differentiate again by filling the "Field" column.

There is not so much work remaining (I don't know about tests) and this pr could be integrated very soon.

**Review by cedric-anne — COMMENTED** [review]
Apart from the few minor comments I made, it seems to me to be OK from the technical point of view.

A more generic system for the translation storage may be introduced by #18223 , but migrating to it, if it is relevant, could be done later and should not be consider as a blocker for the current PR.

**Review by orthagh — APPROVED** [review]
It's still ok for me from a technical pov.
what do you prefer to merge first @cconard96: this pr or the one for reordering fields ?

File: src/Glpi/CustomObject/AbstractDefinition.php [hunk 1]
```diff
@@ -296,11 +328,17 @@ protected function showTranslationForm(): void
                 'item' => $this,
                 'classname' => $this->getCustomObjectClassName(),
                 'translations' => $translations,
+                'fields_dropdown' => Dropdown::showFromArray('field', $custom_fields, [
+                    'display'             => false,
+                    'display_emptychoice' => true,
+                    'emptylabel'          => __('Asset name'),
```
[the remark this answers is not in the document]
Comment (cconard96) [reply src/Glpi/CustomObject/AbstractDefinition.php:334]: How would we determine if it is the asset name or a field? We would have to reserve a name like "asset_name" and ensure no fields can be named the same. Makes more sense to treat an empty value as "no field"/"asset name".

[the remark this answers is not in the document]
Comment (cedric-anne) [reply src/Glpi/CustomObject/AbstractDefinition.php:334]: Values corresponds to custom fields IDs, not to their key, so there would be no ambiguity possible if you use `-1`, `asset_name` or whatever value that is not an integer higher than 0.

**Comment by cconard96** [comment]
> It's still ok for me from a technical pov. what do you prefer to merge first @cconard96: this pr or the one for reordering fields ?

I don't think it matters. This one will probably be approved before the custom field reordering one.

**Comment by trasher** [comment]
I guess tests should be added, at least for new PHP classes/methods.

**Comment by trasher** [comment]
Is this one still WIP? (if not, please remove label)

**Review by cedric-anne — APPROVED** [review]

**Review by AdrienClairembault — APPROVED** [review]

**Pull request merged**